siliconcompiler/server.py

```diff
@@ -313,39 +249,16 @@ async def handle_check_progress(self, request):
         information about the job's progress and intermediary outputs.
         '''
 
-        # Retrieve the JSON parameters.
-        params = await request.json()
-        if not 'job_hash' in params:
-            return web.Response(text="Error: no job hash provided.")
-        job_hash = params['job_hash']
-        if not 'job_id' in params:
-            return web.Response(text="Error: no job ID provided.")
-        jobid = params['job_id']
-        username = ''
-        if 'username' in params:
-            username = params['username']
+        # Process input parameters
+        job_params, response = self.__handle_request(await request.json())
+        if response is not None:
+            return response
 
-        # Check for authentication parameters.
-        use_auth = False
-        if ('username' in params) or ('key' in params):
-            if self.cfg['auth']['value'][-1]:
-                if ('username' in params) and ('key' in params):
-                    username = params['username']
-                    key = params['key']
-                    if not username in self.user_keys.keys():
-                        return web.Response(text="Error: invalid username provided.", status=404)
-                    # Authenticate the user.
-                    if self.auth_password(username, key):
-                        use_auth = True
-                    else:
-                        return web.Response(text="Authentication error.", status=403)
-                else:
-                    return web.Response(text="Error: some authentication parameters are missing.", status=400)
-            else:
-                return web.Response(text="Error: authentication parameters were passed in, but this server does not support that feature.", status=500)
+        job_hash = job_params['job_hash']
+        username = job_params['username']
 
         # Determine if the job is running.
-        if "%s%s_%s"%(username, job_hash, jobid) in self.sc_jobs:
+        if "%s%s"%(username, job_hash) in self.sc_jobs:
```

Comment:
Some of these formatting strings are a bit old; I think the preferred way to format strings in Python 3 is `f"{username}{job_hash}"` instead of the old `"%s%s"%(username, job_hash)` notation.

Reply:
I have a followon PR (not open yet) that fixes this